### PR TITLE
Add multi-step popover for default view components

### PR DESCRIPTION
**Pull request by oisincoveney**
Adds a multi-step popover for Summarize, Breakout, and most likely the new Filter component as well.

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/components/private/util/MultiStepPopover/MultiStepPopover.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        .find(
+          child => isStepElement(child) && child.props.value === currentStep,
+        )?.props.children || null
+    );
+  };
+
+  return (
+    <Popover>
+      <Popover.Target>
+        {Children.toArray(children)
```
Comment (oisincoveney) [inline enterprise/frontend/src/embedding-sdk/components/private/util/MultiStepPopover/MultiStepPopover.tsx:53]: this thing and `getCurrentStepContent` can be consolidated into one utility function, will do that when i get a chance

File: enterprise/frontend/src/embedding-sdk/components/private/util/MultiStepPopover/MultiStepPopover.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+}: PropsWithChildren<{
+  currentStep: string | number;
+}>) => {
+  const getCurrentStepContent = () => {
+    return (
+      Children.toArray(children)
+        .filter((child): child is ReactElement => isValidElement(child))
+        .find(
+          child => isStepElement(child) && child.props.value === currentStep,
+        )?.props.children || null
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/private/util/MultiStepPopover/MultiStepPopover.tsx:46]: I was thinking if there is [an alternative to Children.toArray](https://react.dev/reference/react/Children#alternatives) for this, but given the API this seems to be the best way to go.

File: enterprise/frontend/src/embedding-sdk/components/private/util/MultiStepPopover/MultiStepPopover.unit.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      </MultiStepPopover.Target>
+      <MultiStepPopover.Step value="step1">
+        Step 1 content
+      </MultiStepPopover.Step>
+      <MultiStepPopover.Step value="step2">
+        Step 2 content
+      </MultiStepPopover.Step>
+      <MultiStepPopover.Step value="step3">
+        Step 3 content
+      </MultiStepPopover.Step>
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/private/util/MultiStepPopover/MultiStepPopover.unit.spec.tsx:22]: I like the API - feels very Mantine-y. This kinda emulates the slot pattern you see in Vue/Svelte, which is nice.

**Review by heypoom — APPROVED** [review]
The shape of the API looks nice! Looks like this component is private which is perfect.

File: enterprise/frontend/src/embedding-sdk/components/private/util/MultiStepPopover/MultiStepPopover.unit.spec.tsx (hunk 3 above)
Comment (oisincoveney) [reply enterprise/frontend/src/embedding-sdk/components/private/util/MultiStepPopover/MultiStepPopover.unit.spec.tsx:22]: I'm kinda falling in love with the namespace component setup 😄

**Pull request merged**